src/SFML/Graphics/Shader.cpp

```diff
@@ -60,8 +60,16 @@ sfShader* sfShader_createFromFile(const char* vertexShaderFilename, const char*
         }
         else
         {
-            // vertex + geometry + fragment shaders
-            success = shader->This.loadFromFile(vertexShaderFilename, geometryShaderFilename, fragmentShaderFilename);
+            if (!vertexShaderFilename && !fragmentShaderFilename)
+            {
+                // geometry shader only
+                success = shader->This.loadFromFile(geometryShaderFilename, sf::Shader::Geometry);
+            }
+            else
```

Comment:
What about the situation where either vertexShaderFilename or fragmentShaderFilename is not set, i.e. shouldn't there be three cases as above?

Reply:
I didn't implemented that because SFML doesn't propose the case with geometryShaderFilename + [vertexShaderFilename/fragmentShaderFilename].
To load every case, we should have 7 possibilities.

Reply:
Yes, the C++ code does not support all combinations. To be honest, I think we've reached the limits of the current design of Shader::loadFromXxx functions; if it was written now, I would do it differently.

Reply:
Not sure I fully follow. It's not possible with a single loadFromFile call, but if you want, you can call `loadFromFile(filename, type)` and get all com combinations. So CSFML should allow all combinations as well, no?

Reply:
See [
3f3d623 ](https://github.com/SFML/CSFML/pull/146/commits/3f3d623872b6085cdd3cc25cc9c5e1e06f379b0d) ?

Reply:
> It's not possible with a single loadFromFile call, but if you want, you can call loadFromFile(filename, type) and get all combinations

No no no, this overload loads a single component *and resets the two others*. When geometry shaders didn't exist, it was used to create a program with either a vertex or a fragment shader *only*.

Reply:
Ah okay, I see now. So this still leaves the questions:

- Should CSFML support all the cases?
- Or should CSFML only support the cases SFML supports?
- Should SFML be updated to also support all cases?

Reply:
> Should CSFML support all the cases?

How could it support cases that are not implemented in the C++ API? 😁 

> Should SFML be updated to also support all cases?

That sounds like he wisest solution. Unless some combinations don't make sense?